### PR TITLE
issue-1301 Create an NpmTestRepository annotation

### LINKED ISSUE #1301: Create an NpmTestRepository annotation
# Task Description

As part of our on-going efforts to re-work our tests, so that they're simpler and easier to parallelize an in continuation of our recent work on getting test annotations for the easier generation of test repositories and artifacts, we need to implement an NPM-specific annotation for test repositories. Please, refer to the [documentation](https://strongbox.github.io/developer-guide/writing-tests.html) for more details on how this is implemented. At the moment, we only have these annotations for Maven.

# Tasks

* [ ] Implement an `NpmTestRepository` annotation.
* [ ] Update the [documentation](https://strongbox.github.io/developer-guide/writing-tests.html).

# Task Relationships

This task:
* Relates to: #1165, #1198 , #1219 , #1302 

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by LeifBrooks**
Fixes #1301 .

### CODE REVIEW

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@Documented
+@Retention(RUNTIME)
+@Target({PARAMETER, ANNOTATION_TYPE})
+@TestRepository(layout = NpmArtifactCoordinates.LAYOUT_NAME)
+public @interface NpmTestRepository
+{
+
+    @AliasFor(annotation = TestRepository.class)
+    String storageId() default "storage0";
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java]: I think the default for this should be `storage-npm`.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    @AliasFor(annotation = TestRepository.class)
+    String repositoryId() default "releases";
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java]: I think the default value for this should be `npm-releases`.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java (hunk 1 above)
Comment (LeifBrooks) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java]: Updated

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java (hunk 2 above)
Comment (LeifBrooks) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/testing/repository/NpmTestRepository.java]: Updated

**Comment by carlspring** [comment]
Thanks for this! :)

**Pull request merged**